### PR TITLE
Clean up s_server documentation

### PR DESCRIPTION
**Pull request by mattcaswell**
List the options in the same order and in the same style as the output from
"openssl s_server -help"

There are still a number of inconsistencies here. For example some options are not documented here at all, but we refer to another page (SSL_CONF_cmd) where they are documented. Other options are documented on both pages. I'll deal with that separately.

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man1/s_server.pod [hunk 1]
```diff
@@ -8,116 +8,170 @@ s_server - SSL/TLS server program
 
 B<openssl> B<s_server>
 [B<-help>]
-[B<-port port>]
+[B<-port +int>]
```
Comment (richsalz) [inline doc/man1/s_server.pod]: why the plus before int?

File: doc/man1/s_server.pod [hunk 2]
```diff
@@ -166,23 +212,58 @@ Use IPv4 only.
 
 Use IPv6 only.
 
-=item B<-context id>
+=item B<-unlink>
+
+For -unix, unlink existing socket first.
```
Comment (kaduk) [inline doc/man1/s_server.pod]: Maybe "any existing socket"?

File: doc/man1/s_server.pod [hunk 3]
```diff
@@ -166,23 +212,58 @@ Use IPv4 only.
...
+
+=item B<-context val>
 
 Sets the SSL context id. It can be given any string value. If this option
 is not present a default value will be used.
 
-=item B<-cert certname>
+=item B<-verify int>, B<-Verify int>
```
Comment (kaduk) [inline doc/man1/s_server.pod]: Wow, we really have a case-dependent long-form argument like that.  Not that we could change it until 1.2, of course...

File: doc/man1/s_server.pod (hunk 1 above)
Comment (mattcaswell) [inline doc/man1/s_server.pod]: It is the convention (that you introduced) used in the apps to denote that an argument expects an integer greater than 0. If you do `openssl s_server -help` that is the help you get:

    $ ../util/shlib_wrap.sh ./openssl s_server -help
    Usage: s_server [options]
    Valid options are:
     -help                      Display this summary
     -port +int                 TCP/IP port to listen on for connections (default is 4433)

I'm just trying to remain consistent.

Comment (richsalz) [inline doc/man1/s_server.pod]: I did?  Wow, okay. :)

File: doc/man1/s_server.pod (hunk 2 above)
Comment (mattcaswell) [inline doc/man1/s_server.pod]: Fixed.

File: doc/man1/s_server.pod (hunk 3 above)
Comment (mattcaswell) [inline doc/man1/s_server.pod]: Unfortunately, yes.

**Comment by mattcaswell** [comment]
@kaduk, updated commit pushed addressing your comment.

File: doc/man1/s_server.pod [hunk 4]
```diff
@@ -8,116 +8,170 @@ s_server - SSL/TLS server program
...
-[B<-verify_ip ip>]
-[B<-verify_name name>]
-[B<-x509_strict>]
-[B<-nocert>]
-[B<-client_sigalgs sigalglist>]
-[B<-named_curve curve>]
-[B<-cipher cipherlist>]
-[B<-serverpref>]
-[B<-quiet>]
-[B<-ssl3>]
```
Comment (kaduk) [inline doc/man1/s_server.pod]: Is it intentional to leave -ssl3 out of the SYNOPSIS (but leave it in the body description?

Comment (mattcaswell) [inline doc/man1/s_server.pod]: Oh, err no. I created this SYNOPSIS directly from the -help output - but of course ssl3 is disabled in the default build. I'll add it back.

Comment (mattcaswell) [inline doc/man1/s_server.pod]: Fixed

**Comment by mattcaswell** [comment]
Updated commit pushed.

**Comment by mattcaswell** [comment]
@kaduk, hmmm looks like you've not been added to the addrev tool yet, so your "Reviewed-by" header added by hand. Pushed, anyway. Thanks.

**Pull request closed without merging**